### PR TITLE
API Increase minimum PHP version to 7.1, include PHPUnit 7

### PR DESCRIPTION
**Pull request by robbieaverill**
See silverstripe/silverstripe-framework#7653

Requires:

- [x] silverstripe/silverstripe-framework#7716

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -3,13 +3,13 @@
...
     "require": {
-        "php": ">=5.6.0",
+        "php": ">=7.1.0",
         "silverstripe/recipe-plugin": "^1",
         "silverstripe/recipe-cms": "2.x-dev",
         "silverstripe-themes/simple": "~3.2.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "^5.7"
+        "phpunit/phpunit": "7.0.x-dev"
```
Comment (tractorcow) [inline composer.json]: `^7dev` please.

File: .travis.yml [hunk 2]
```diff
@@ -66,7 +66,7 @@ before_script:
 # Install composer dependencies
   - composer validate
   - if [[ $DB == PGSQL ]]; then composer require --no-update silverstripe/postgresql:2.0.x-dev; fi
-  - if [[ $BEHAT_TEST ]]; then composer require --no-update silverstripe/behat-extension:^3 silverstripe/serve:dev-master se/selenium-server-standalone:2.41.0; fi;
+  - if [[ $BEHAT_TEST ]]; then composer require --no-update silverstripe/behat-extension:4.x-dev silverstripe/serve:dev-master se/selenium-server-standalone:2.41.0; fi;
```
Comment (tractorcow) [inline .travis.yml]: Let's merge the behat-extension PR first, and we'll tag a 4.0.0-alpha1. Then let's update this to use the stable tag over the dev branch.

**Comment by tractorcow** [comment]
You may need to wait until https://github.com/silverstripe/silverstripe-installer/pull/214 is merged and then merged up.

**Comment by robbieaverill** [comment]
Rebased. We're getting closer! The two failures in the `core` test suite are fixed with https://github.com/silverstripe/silverstripe-assets/pull/106, the failure in the `admin` test suite look like they could be real.

**Comment by dhensby** [comment]
@robbieaverill I've merged asset-admin up for you

**Comment by robbieaverill** [comment]
Ok, still failures there. Could be affected by global state, or vice versa (I'm guessing now).

**Comment by tractorcow** [comment]
Restarted tests.

**Comment by dhensby** [comment]
failure in admin :/

**Comment by tractorcow** [comment]
Pre-existing; Good enough to merge.

**Pull request merged**